### PR TITLE
Feature/compartment

### PR DESCRIPTION
**Pull request by sbhagate-infoblox**
Modified protoc-gen-gorm to support resource compartment.

### CODE REVIEW

File: example/postgres_arrays/postgres_arrays.pb.gorm.go [hunk 1]
```diff
@@ -3,9 +3,9 @@ package postgres_arrays
 import (
 	context "context"
 	fmt "fmt"
-	gateway "github.com/infobloxopen/atlas-app-toolkit/gateway"
 	errors "github.com/infobloxopen/protoc-gen-gorm/errors"
 	pq "github.com/lib/pq"
+	gateway "github.com/sbhagate-infoblox/atlas-app-toolkit-1.4.0/gateway"
```
Comment (kumaya) [inline example/postgres_arrays/postgres_arrays.pb.gorm.go:8]: revert this

File: plugin/plugin.go [hunk 2]
```diff
@@ -2370,7 +2431,16 @@ func (b *ORMBuilder) generatePatchHandler(message *protogen.Message, g *protogen
 		isMultiAccount = true
 	}
 
-	if isMultiAccount && !b.hasIDField(message) {
+	if getMessageOptions(message).GetMultiCompartment() {
+		isMultiCompartment = true
+	}
+
+	if isMultiAccount && isMultiCompartment && !b.hasIDField(message) {
```
Comment (kumaya) [inline plugin/plugin.go:2438]: no need for this check

File: plugin/plugin.go [hunk 3]
```diff
@@ -2480,14 +2549,22 @@ func (b *ORMBuilder) generateAfterPatchHookDef(orm *OrmableType, suffix string,
...
 
 func (b *ORMBuilder) generatePatchSetHandler(message *protogen.Message, g *protogen.GeneratedFile) {
-	var isMultiAccount bool
+	var isMultiAccount, isMultiCompartment bool
 
 	typeName := string(message.Desc.Name())
 	if getMessageOptions(message).GetMultiAccount() {
 		isMultiAccount = true
 	}
+	if getMessageOptions(message).GetMultiCompartment() {
```
Comment (kumaya) [inline plugin/plugin.go:2558]: same here

File: example/user/user.proto [hunk 4]
```diff
@@ -14,6 +14,7 @@ message User {
     option (gorm.opts) = {
         ormable: true,
         multi_account: true
+        multi_compartment: true
```
Comment (kumaya) [inline example/user/user.proto:17]: add example to just one message

File: testdata/test.proto [hunk 5]
```diff
@@ -0,0 +1,105 @@
+syntax = "proto3";
```
Comment (kumaya) [inline testdata/test.proto:1]: you can update the existing example instead of creating new

File: plugin/plugin.go [hunk 6]
```diff
@@ -36,7 +36,7 @@ var (
 	gormImport         = "gorm.io/gorm"
 	tkgormImport       = "github.com/infobloxopen/atlas-app-toolkit/gorm"
 	uuidImport         = "github.com/satori/go.uuid"
-	authImport         = "github.com/infobloxopen/atlas-app-toolkit/auth"
+	authImport         = "github.com/infobloxopen/protoc-gen-gorm/auth"
```
Comment (drewwells) [inline plugin/plugin.go:39]: 💯

File: go.mod [hunk 7]
```diff
@@ -1,18 +1,26 @@
...
-	github.com/infobloxopen/atlas-app-toolkit v1.2.2-0.20220721202134-28480c4993fc
-	github.com/jackc/pgtype v1.11.0 // indirect
+	github.com/grpc-ecosystem/go-grpc-middleware v1.2.2
+	github.com/infobloxopen/atlas-app-toolkit v1.4.0
 	github.com/jinzhu/inflection v1.0.0
 	github.com/lib/pq v1.10.2
+	github.com/mattn/go-sqlite3 v1.14.12 // indirect
 	github.com/satori/go.uuid v1.2.0
-	go.opencensus.io v0.23.0
+	go.opencensus.io v0.22.4
```
[the remark this answers is not in the document]
Comment (sbhagate-infoblox) [reply go.mod]: It is updated to go.opencensus.io v0.23.0

File: testdata/test.proto (hunk 5 above)
Comment (sbhagate-infoblox) [reply testdata/test.proto:1]: Daniel-garcia suggested to add test case.
I added a test case in a similar manner done in atlas-gentool.

Comment (kumaya) [reply testdata/test.proto:1]: Can you point me to test cases ?

Comment (sbhagate-infoblox) [reply testdata/test.proto:1]: test case added in testdata folder.

**Pull request closed without merging**